### ISSUE TITLE
Lint implicit return assignments in `function_return_linter()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [jimhester/lintr#1574]
Follow-up on https://github.com/r-lib/lintr/pull/1569

Extend `function_return_linter()` to also lint implicit return assignments like the following (cf. https://github.com/r-lib/lintr/pull/1569#discussion_r984199472):

``` r
foo <- function() {
  x <- 4
}
y <- foo() 
y
#> [1] 4

library(lintr)

lint(
  text = "foo <- function() {
            x <- 4
          }",
  linters = function_return_linter()
)
```

<sup>Created on 2022-09-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
Should `function_return_linter()` be subsumed into `return_linter()`?

**Comment by IndrajeetPatil** [comment]
Yes, I think that's a good idea.
